src/proxy-ui-api/src/main/java/org/niis/xroad/securityserver/restapi/service/ServiceDescriptionService.java

```diff
@@ -885,12 +885,16 @@ public String getServiceTitle(ClientType clientType, String serviceCode) {
         ServiceType service = clientType.getServiceDescription().stream()
                 .flatMap(sd -> sd.getService().stream())
                 .filter(serviceType -> serviceType.getServiceCode().equals(serviceCode))
-                .findFirst()
+                .max(comparingVersions())
                 .orElse(null);
 
         return service == null ? null : service.getTitle();
     }
 
+    private Comparator<ServiceType> comparingVersions() {
+        return Comparator.comparing(service -> Integer.valueOf(service.getServiceVersion().replace("v","")));
+    }
+
```

Comment:
But I understand the problem this pull request tries to solve and I think we are able to solve it with small modifications. Instead of counting on specifically formatted version numbers, we could do a reverse alphabetical comparison. It should produce exactly the same effect.

Reply:
I would put in some more test data to the `ServiceDescriptionServiceIntegrationTest#getServiceTitle` test and ensure that it behaves as expected with version strings such as `AAA`, `XYZ`, `v1`, V1`. Once that works, I'm sure we can proceed with integrating the change to the X-Road core.